src/templates/next/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from "react"
+import { BiChevronDown } from "react-icons/bi"
+import { FilterProps } from "~/common"
+import { Heading } from "../../typography/Heading"
+
+const Filter = ({
+  filters,
+  appliedFilters,
+  setAppliedFilters,
+}: Omit<FilterProps, "type">) => {
+  const [showFilter, setShowFilter] = useState<Record<string, boolean>>({})
```

Comment:
does `useState` still work with server sided react?

Reply:
Yup

Reply:
actually im abit confused on this front
https://nextjs.org/learn/react-foundations/server-and-client-components

I thought it wouldn't work since next uses server components by default, but I also don't see anywhere in our `isomer-next-base-template` that we use the `use client` directive

can fill me in on what im missing here @dcshzj

Reply:
Yup, it is mainly because we are using the pages router, which [renders components on the client side by default](https://www.wearedevelopers.com/magazine/next-js-app-router-explained). In contrast, the app router renders components on the server side by default, hence the need for `use client` in that scenario. Because the default is already client-side, so we don't need to add this directive at all in our components.

So when we are using SSG, we are building the pages on the server and [rendering as much as we can on the server](https://stackoverflow.com/a/75550697). Then, the HTML and CSS, along with the related JS content needed for client-side functionality to work, will be sent to the user. Hence, you will see the site with the proper HTML sent correctly to the user, but there will still be some client-side actions that are performed using the JS executed on the user's browser. The `useState` hook is one of them.

The server-side of things comes in under the `getStaticProps` function, so it knows what props to pass into the components to render as much as possible on the server. The `getStaticPaths` is the other function to know which routes the server needs to "access" to render for. This just leaves places where React hooks (which are client-side) are used, and that is then offloaded to the user's browser to execute.

So my "yup" earlier was referring to the fact that `useState` will still work regardless, but this is still done on the client side and not rendered on the server.